tests/test_graph.py

```diff
@@ -158,6 +158,27 @@ def test_disconnected_graph(self) -> None:
         path = g1.find_shortest_path(v1, v3)
         self.assertIsNone(path)
 
+    def test_dijkstra_search(self) -> None:
```

Comment:
Since this is a fairly critical core algorithm I suggest adding a few more tests:
* edge case graph with only one vertex and no edges: make sure the algorithm doesn't break
* simple triangle graph (with vertexes v1, v2, v3 and edges/weights v1->v2: 1, v2->v3: 2, v1->v3: 4): find shortest path from v1 to v3
* same as previous case but with v1->v3: 2.
* same as previous case but with v1->v3: 3.
* graph with a cycle (i.e. v2 -> v3, v3-> v4, v4 -> v2)
* also worth exploring: what happens if  the graph is made out of 2 disconnected components (i.e. two "island" graphs that are not connected by any edge? Just make sure the algorithm returns sane results (e.g. empty list or similar)
* more variants on the graph you added. meaning try to find the shortest path between other vertexes in the same graph
* any other case you may want to add

Reply:
To be clear, this is a one way algorithm and cyclic so in case 3:

> same as previous case but with v1->v3: 2.

The route should still be `[v3, v2, v1]` since the link should not be reciprocal.

Reply:
It's true that Graph is unidirectional. Here are the expected results for cases 2 and 3:
* case 2: there are two possible paths: v1 -> v3 and v1 -> v2 -> v3. The result should be: v1 -> v2 -> v3 (because the total weight is 3 as opposed to the other path which has weight 4).
*  case 3: there are two possible paths: v1 -> v3 and v1 -> v2 -> v3. The result should be v1 -> v3 (because the total weight is 2 as opposed to the other path which has weight 3)

Reply:
> never mind, read that wrong. Wasted a lot of time as a result. This is making my brain hurt. haha

Haha, that happens! I recommend actually drawing the graphs on paper so that you can reason intuitively: you can both follow the algorithm step by step on a real graph and tell what the right path is.